src/shared_modules/content_manager/src/actionOrchestrator.hpp

```diff
@@ -36,7 +36,31 @@ class ActionOrchestrator final
     enum UpdateType
     {
         CONTENT,
-        OFFSET
+        OFFSET,
+        FILE_HASH
+    };
+
+    /**
+     * @brief Struct containing the necessary members to execute the orchestrations.
+     *
+     * @details When 'type' is CONTENT, the orchestrator will perform a content update. If 'offset' is
+     * zero, the offset value will be reset. Otherwise, its value will be read from the RocksDB database.
+     * @details When 'type' is OFFSET, the orchestrator will write the 'offset' value in the RocksDB database. This
+     * value should be nonnegative.
+     * @details When 'type' is FILE_HASH, the orchestrator will write the 'fileHash' value in the RocksDB database. This
+     * value shouldn't be empty.
+     */
+    struct UpdateData
+    {
+        UpdateType type {UpdateType::CONTENT}; ///< Orchestration update type.
+        int offset {1};                        ///< Offset value used in the update.
+        std::string fileHash;                  ///< Hash value used in the update.
+
+        /**
+         * @brief Struct empty constructor.
+         *
+         */
+        UpdateData() {};
```

Comment:
If the members of the struct have default values, then I think that the empty default constructor does not make much sense.
It could be removed. Or take another approach and do not set default values to the member fields and create a constructor where they must be set explicitly.

Reply:
To help enforce all the conditions, how about:
- Make a private constructor with all the parameters
- Add one static function for each type of update, check that all the parameters are valid before calling the constructor and returning the UpdateData object:
For example:
```
static UpdateData createHashUpdate(const std::string& hash) {
        if(hash.empty()){
         throw...
        }
        return UpdateData(UpdateType::FileHash, hash, -1);
    }
```

Reply:
So, I've chosen @jftuduri 's approach. Seems more secure than creating the struct directly, avoiding so the introduction of unwanted values.